adafruit_turtle.py

```diff
@@ -30,9 +31,13 @@
 import gc
 import math
 import time
-import board
 import displayio
 
+try:
+    import board
```

Comment:
This is an usual import to mask, both because it seems its used functionally in this module (not just for typing) and because it silently does so, masking what should be an exception with a `print()` statement.  Is there any reason when we're catching `NotImplementedError` instead of `ImportError`?  Maybe I'm missing something, I still need to check the rest, but wanted to ask about this in the meantime.

Reply:
Hey @tekktrik, the reason I catch that exception is so that I can run this module on systems that don't have the `board` module. When running with Blinka, the board module is defined for systems such as a Raspberry Pi SBC, but on generic x86 computers, the Blinka library will raise a `NotImplementedError` error (not sure why it's not an `ImportError`). That exception is thrown at the bottom of [this](https://github.com/adafruit/Adafruit_Blinka/blob/main/src/board.py) file.

I suppose another approach could be to remove the import all together, and then in the `turtle` initializer make the `display` argument a required one. This would force users to pass the `board.DISPLAY` for CircuitPython boards with a built-in display, vs the current behavior where that is the default -- note that this itself is wrapped in a try/except, which seems a bit weird because it will cause an exception on CircuitPython boards without a built-in display. This change, however, would break some existing scripts that use the library without passing the `display` parameter.

Let me know what you think!

Reply:
I think this was written before the generic x86 computer implementation was added, hence why it currently only catches `AttributeError`.  However, is there a use case where a display can be defined for a generic x86 computer?  I'm not familiar with that implementation of Blinka and how it's used, and I've reached out to the other devs as well.  If you know of one, it would help me out!

Reply:
Yes there is! @FoamyGuy has one [here](https://github.com/FoamyGuy/Blinka_Displayio_PyGameDisplay), and I've included that code in my project [here](https://github.com/shulltronics/displayio-universal-ui). It's fun because you can incorporate `displayio` graphics into CPython applications. It's also useful for designing/testing embedded UIs for CircuitPython when you might not have access to the hardware.

I've been using this to demonstrate the power of abstraction/modularity to some of my students at University of Florida, where we can run (almost) identical code on laptops, RPis (either with HDMI or a PiTFT), and on microcontrollers with an embedded display -- just by changing the `display` object to target the proper display for our system.

I worked on this pull request because I want to be able to use `turtle` in a similar fashion, but the current implementation just doesn't allow for it.

Reply:
Got it, thanks!  I'll get to this in the next few days, but that helps but the change into context!

Reply:
Another potential option is to move the import down to inside the if statement so that it will only try to import board. if the user has not passed a display object.